_posts/2020-04-17-quarkus-mutual-tls.adoc

```diff
@@ -0,0 +1,405 @@
+---
+layout: post
+title: 'Quarkus - Makes Mutual TLS Easy.'
+date:   2020-04-17 00:00 +0100
+author: mkouba
+synopsis: Quarkus - Mutual TLS Authentication.
+---
+
+== What is Mutual TLS Authentication?
+
+Mutual TLS authentication or two way authentication is an extension of Transport layer security (or “TLS”), and it ensures that traffic between the client and server is secure and trusted in both directions.
+
+== Why Mutual TLS?
+
+It is an additional security layer which verifies the identity of the client.
+
+== How implement it?
+
+The best approach to implement Mutual TLS between two services has to come from the infrastructure, for instance, Service Mesh, to achieve a standard and secure way of communication and to avoid that each application implements their own solution. However, you are not always working on cutting edge environment.
+
+Let's have a look at how to implement mTLS with Quarkus.
+ 
+== Bootstrapping.
+
+Bootstrapping Hello server application,
+
+[source]
+----
+mvn io.quarkus:quarkus-maven-plugin:1.4.0.CR1:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=quarkus-server-mtls \
+    -DclassName="org.acme.server.mtls.GreetingResource" \
+    -Dpath="/hello" \
+    -DplatformArtifactId=quarkus-bom \
+    -DplatformVersion=1.4.0.CR1
```

Comment:
DplatformVersion won't be needed if you use a Final version

Reply:
Correct